addons/account_asset/account_asset_invoice.py

```diff
@@ -46,6 +46,9 @@ def asset_create(self, cr, uid, lines, context=None):
         context = context or {}
         asset_obj = self.pool.get('account.asset.asset')
         for line in lines:
+            asset_id = asset_obj.search(cr, uid, [('code', '=', line.invoice_id.number)], context=context)
```

Comment:
on account.invoice, check move_id and compare to account_move_ids on account.assets.assets